src/coreclr/src/vm/jitinterface.cpp

```diff
@@ -9002,33 +9009,36 @@ CORINFO_METHOD_HANDLE CEEInfo::resolveVirtualMethodHelper(CORINFO_METHOD_HANDLE
         Assembly* pCallerAssembly = callerMethod->GetModule()->GetAssembly();
         bool allowDevirt =
             IsInSameVersionBubble(pCallerAssembly , pDevirtMD->GetModule()->GetAssembly())
-            && IsInSameVersionBubble(pCallerAssembly , pDerivedMT->GetAssembly());
+            && IsInSameVersionBubble(pCallerAssembly, pExactMT->GetAssembly());
```

Comment:
Not sure if this change makes sense; for an instantiation which assembly is returned?

Assuming this does make sense, if the exact class would block devirt we might try checking again with the shared class...?

Reply:
`GetAssembly` is not affected by instantiation, but this change loses track of the R2R protection. Instead of using `pExactMT` should be `pObjMT`

Reply:
Ah, right. We need to check that the full inheritance chain is in the bubble.